### PR TITLE
[#383] Upgrade Node and Ruby dependency versions

### LINKED ISSUE #383: Upgrade Gem versions
## Why

As we are now in 2023, we should revisit some Gem version, for example the `gem 'rspec-rails', '~> 5.0.2'`, the latest version is 6.x.x
 
## Who Benefits?

Developer

### PR DESCRIPTION
**Pull request by andyduong1920**
close #383

## What happened 👀

Upgrade Node and Ruby dependency versions

## Insight 📝

- Still stay as Node 16, there is a [new issue for Node 18](https://github.com/nimblehq/rails-templates/issues/421)
- I changed to use the new `npm pkg set` syntax over the `npm set-script` ([Slack message](https://nimble-co.slack.com/archives/C01GL6RBW11/p1688357829582519)) on this PR too

## Proof Of Work 📹

The CI is passed

### CODE REVIEW

File: Gemfile.tt [hunk 1]
```diff
@@ -53,7 +53,7 @@ group :development, :test do
   # Code Analysis
   gem 'bullet' # help to kill N+1 queries and unused eager loading
   gem 'brakeman', require: false # A static analysis security vulnerability scanner for Ruby on Rails applications
-  gem 'parser', '3.0.1.1' # Use correct parser version to avoid parser warnings
+  gem 'parser', '3.2.2.3' # Use correct parser version to avoid parser warnings
```
Comment (olivierobert) [inline Gemfile.tt:56]: Do we know why we need this gem?

**Review by olivierobert — APPROVED** [review]
Just have one small question, but the PR looks good

File: Gemfile.tt (hunk 1 above)
Comment (andyduong1920) [reply Gemfile.tt:56]: I didn't have time to check that as it already in the template @olivierobert there are some gem that I think we don't really use it at all but still in the Gemfile.tt, I will create a separate issue to revisit the Template Gemfile.tt 👌

Comment (andyduong1920) [reply Gemfile.tt:56]: Issue created https://github.com/nimblehq/rails-templates/issues/426

**Review by malparty — APPROVED** [review]

**Review by sanG-github — APPROVED** [review]

**Pull request merged**